okhttp/src/test/java/okhttp3/CallHandshakeTest.kt

```diff
@@ -0,0 +1,197 @@
+      .hostnameVerifier(RecordingHostnameVerifier())
+      .build()
+    server.useHttps(handshakeCertificates.sslSocketFactory(), false)
+
+    defaultEnabledCipherSuites = handshakeCertificates.sslSocketFactory().defaultCipherSuites.toList()
+    defaultSupportedCipherSuites = handshakeCertificates.sslSocketFactory().supportedCipherSuites.toList()
+  }
+
+  @Test
+  fun testDefaultHandshakeCipherSuiteOrderingTls12Restricted() {
+    val client = makeClient(ConnectionSpec.RESTRICTED_TLS, TlsVersion.TLS_1_2)
+
+    val handshake = makeRequest(client)
+
+    assertThat(handshake.cipherSuite).isIn(expectedModernTls12CipherSuites)
+
+    // Probably something like
+    // TLS_AES_128_GCM_SHA256
+    // TLS_AES_256_GCM_SHA384
+    // TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384
+    // TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
+    // TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
+    // TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
+    assertThat(handshakeEnabledCipherSuites).containsExactlyElementsOf(expectedConnectionCipherSuites(client))
+  }
+
+  @Test
+  fun testDefaultHandshakeCipherSuiteOrderingTls12Modern() {
+    val client = makeClient(ConnectionSpec.MODERN_TLS, TlsVersion.TLS_1_2)
+
+    val handshake = makeRequest(client)
+
+    assertThat(handshake.cipherSuite).isIn(expectedModernTls12CipherSuites)
+
+    // Probably something like
+    // TLS_AES_128_GCM_SHA256
+    // TLS_AES_256_GCM_SHA384
+    // TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384
+    // TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
+    // TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
+    // TLS_RSA_WITH_AES_256_GCM_SHA384
+    // TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
+    // TLS_RSA_WITH_AES_128_GCM_SHA256
+    // TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA
+    // TLS_RSA_WITH_AES_256_CBC_SHA
+    // TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA
+    // TLS_RSA_WITH_AES_128_CBC_SHA
+    assertThat(handshakeEnabledCipherSuites).containsExactlyElementsOf(expectedConnectionCipherSuites(client))
+  }
+
+  @Test
+  fun testDefaultHandshakeCipherSuiteOrderingTls13Modern() {
+    // Requires modern JVM
+    platform.expectFailureOnJdkVersion(8)
+
+    val client = makeClient(ConnectionSpec.MODERN_TLS, TlsVersion.TLS_1_3)
+
+    val handshake = makeRequest(client)
+
+    assertThat(handshake.cipherSuite).isIn(expectedModernTls13CipherSuites)
+
+    // TODO: filter down to TLSv1.3 when only activated.
+    // Probably something like
+    // TLS_AES_128_GCM_SHA256
+    // TLS_AES_256_GCM_SHA384
+    // TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384
+    // TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
+    // TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
+    // TLS_RSA_WITH_AES_256_GCM_SHA384
+    // TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
+    // TLS_RSA_WITH_AES_128_GCM_SHA256
+    // TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA
+    // TLS_RSA_WITH_AES_256_CBC_SHA
+    // TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA
+    // TLS_RSA_WITH_AES_128_CBC_SHA
+    assertThat(handshakeEnabledCipherSuites).containsExactlyElementsOf(expectedConnectionCipherSuites(client))
```

Comment:
this isn’t asserting the order. Is that unexpected?

Reply:
I'm going to land with it more relaxed and passing then we can define exactly what our current and future expectations are.

Reply:
It is asserting the order according to javadoc

```
   * Same as {@link #containsExactly(Object...)} but handle the {@link Iterable} to array conversion : verifies that
   * actual contains exactly the elements of the given iterable and nothing else <b>in the same order</b>.
```